modules/shared/nRF52840/inc/user_part_export.c

```diff
@@ -10,14 +10,14 @@
 /**
  * Locations of static memory regions from linker.
  */
-extern char link_global_data_initial_values;
-extern char link_global_data_start;
-extern char link_global_data_end;
-#define link_global_data_size (&link_global_data_end - &link_global_data_start)
-
-extern char link_bss_location;
-extern char link_bss_end;
-#define link_bss_size (&link_bss_end - &link_bss_location)
+extern uintptr_t link_global_data_initial_values;
```

Comment:
Does GCC with LTO enabled complains about these symbols declared as `char`? Using `uintptr_t` here is confusing since you have to take the address of such a symbol in the code rather than use it by value.

Reply:
Yeah, there was an ODR violation somewhere and we've done a similar change previously when implementing original GCC 9 support, so the intent was to use uintptr_t commonly (https://github.com/particle-iot/device-os/pull/2103#discussion_r425725917)